### ISSUE TITLE
Possible bug in JdoDataStore?

### ISSUE THREAD

**Issue by pedromvu** [googleapis/google-oauth-java-client#453]
I couldn't find the github main source code of this file but found it here: https://www.javatips.net/api/google-http-java-client-master/google-http-client-jdo/src/main/java/com/google/api/client/extensions/jdo/JdoDataStoreFactory.java

I needed it to know how to implement a MongoDB equivalent and noticed something odd in the set method, although it looks like it's intentional, was wondering why it is like that.

```  
     JdoValue jdoValue = executeKeyQuery(query, key);
     if (jdoValue != null) {
         jdoValue.serialize(value);
     } else {
        jdoValue = new JdoValue(getId(), key, value);
        persistenceManager.makePersistent(jdoValue);
     }
```

So when token is refreshed for an already existing key, the new StoredCredential value is not persisted?

Is the data synced automatically somehow?

**Comment by chingor13** [comment]
Note that JdoDataStoreFactory was deprecated and hasn't been available as of v1.29.0 of `com.google.http-clieht:google-http-client`.

To correctly use a `DataStoreFactory` implementation, you will want to register it in a [`CredentialRefreshListener`](https://googleapis.dev/java/google-oauth-client/latest/index.html?com/google/api/client/auth/oauth2/CredentialRefreshListener.html). In this case, a [`DataStoreCredentialRefreshListener`](https://googleapis.dev/java/google-oauth-client/latest/index.html?com/google/api/client/auth/oauth2/DataStoreCredentialRefreshListener.html) something like:
```java
Credential credential = new Credential.Builder(BearerToken.authorizationHeaderAccessMethod())
    .setTransport(transport)
    .setJsonFactory(jsonFactory)
    .setTokenServerUrl(new GenericUrl(tokenServerUrl))
    // should store the credentials when the credentials are refreshed
    .addRefreshListener(new DataStoreCredentialRefreshListener(userId, dataStoreFactory))
    .build();
```

**Comment by pedromvu** [comment]
Thanks, I guess this doesn't solve my question unless the original developer could answer.
